src/operator/mshadow_op.h

```diff
@@ -128,6 +128,27 @@ using std::is_integral;
 
 MXNET_UNARY_MATH_OP_NC(identity, a);
 
+template <typename IType, typename DType>
+struct IndexedNum {
+  IType idx;
+  DType num;
+
+  MSHADOW_XINLINE IndexedNum() : idx(0), num(0) {}
+
+  MSHADOW_XINLINE IndexedNum(DType n) : idx(0), num(n) {}
+
+  MSHADOW_XINLINE IndexedNum& operator+=(const IndexedNum& rhs){
+    return *this;
+  }
+};
+
+template<typename DType, typename OType>
+struct arg_min_max_map : public mxnet_op::tunable {
```

Comment:
why not just identity?

Reply:
Here OType is actually IndexedNum and identity is just 
```
  struct name : public mxnet_op::tunable { 
    template<typename DType> 
    MSHADOW_XINLINE static DType Map(DType a) { 
      return a; 
    } 
```

Reply:
Still, if you made `OType a = identity(x)` it would still work, right?